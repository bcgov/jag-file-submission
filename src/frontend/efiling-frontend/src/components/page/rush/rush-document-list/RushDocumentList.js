import React from "react";
import PropTypes from "prop-types";
import RushDocumentListItem from "./RushDocumentListItem";
import { propTypes } from "../../../../types/propTypes";

import "./RushDocumentList.scss";

const hash = require("object-hash");

export default function RushDocumentList({ files }) {
  return (
    <div className="ct-file-list ">
      <div className="header">
        <span className="d-none d-lg-inline col-lg-6">Document Name</span>
        <span className="d-none d-lg-inline col-lg-3">Action</span>
      </div>
      <ul>
        {files &&
          files.map((file) => (
            <RushDocumentListItem
              file={file}
              key={hash(file)}
            />
          ))}
      </ul>
    </div>
  );
}

RushDocumentList.propTypes = {
  files: PropTypes.arrayOf(propTypes.file.isRequired).isRequired,
};
