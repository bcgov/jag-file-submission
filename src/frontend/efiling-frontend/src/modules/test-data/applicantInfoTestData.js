const applicantInfo = {
  bceid: "bobross42",
  firstName: "Bob",
  lastName: "Ross",
  email: "bob.ross@example.com",
};

export function getApplicantInfo() {
  return applicantInfo;
}
