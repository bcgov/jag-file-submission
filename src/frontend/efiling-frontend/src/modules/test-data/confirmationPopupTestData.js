/* eslint-disable react/jsx-one-expression-per-line */
import React from "react";

const onButtonClick = () => {};

const mainButton = {
  label: "Cancel",
  styling: "bcgov-normal-white btn",
  onClick: onButtonClick,
};

const confirmButton = {
  label: "Yes, cancel",
  styling: "bcgov-normal-blue btn bcgov-consistent-width",
  onClick: onButtonClick,
};

const cancelButton = {
  label: "No, dont cancel",
  styling: "bcgov-normal-white btn bcgov-consistent-width",
  onClick: onButtonClick,
};

const body = () => (
  <>
    <p>Your files will not be submitted.</p>
    <p>
      You will be returned to:
      <br />
      <b>Original</b> website
    </p>
  </>
);

export function getTestData() {
  const modal = {
    show: false,
    title: "title",
    body,
  };

  return {
    modal,
    mainButton,
    confirmButton,
    cancelButton,
  };
}
