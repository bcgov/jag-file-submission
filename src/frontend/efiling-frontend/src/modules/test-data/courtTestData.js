const court = {
  locationDescription: "Court location",
  fileNumber: "Court file number",
  levelDescription: "Level",
  classDescription: "Class",
  level: "P",
  courtClass: "F",
  location: "Kelowna Law Courts",
};

export function getCourtData() {
  return court;
}
