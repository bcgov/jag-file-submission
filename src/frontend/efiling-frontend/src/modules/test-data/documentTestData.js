const documents = [
  {
    documentProperties: {
      name: "file name 1",
      type: "file type",
    },
    description: "file description 1",
    statutoryFeeAmount: 40,
    isAmendment: null,
    isSupremeCourtScheduling: null,
  },
  {
    documentProperties: {
      name: "file name 2",
      type: "file type",
    },
    description: "file description 2",
    statutoryFeeAmount: 0,
    isAmendment: false,
    isSupremeCourtScheduling: true,
  },
];

export function getDocumentsData() {
  return documents;
}
