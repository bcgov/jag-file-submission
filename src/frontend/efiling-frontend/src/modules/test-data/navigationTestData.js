const navigationUrls = {
  cancel: "cancelurl.com",
  success: "successurl.com",
  error: "",
};

export function getNavigationData() {
  return navigationUrls;
}
