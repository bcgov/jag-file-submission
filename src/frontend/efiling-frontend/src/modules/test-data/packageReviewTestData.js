export const documents = [
  {
    identifier: "1",
    documentProperties: {
      type: "AFF",
      name: "test-document.pdf",
    },
    status: {
      description: "Submitted",
    },
    filingDate: "2020-05-05T00:00:00.000Z",
  },
];

export const parties = [
  {
    partyType: "IND",
    roleType: "APP",
    firstName: "Bob",
    middleName: "Q",
    lastName: "Ross",
    roleDescription: "Applicant",
    partyDescription: "Individual",
  },
];

export const payments = [
  {
    feeExempt: false,
    paymentCategory: 1,
    processedAmount: 7,
    submittedAmount: 7,
    serviceIdentifier: 0,
    transactionDate: null,
  },
];
