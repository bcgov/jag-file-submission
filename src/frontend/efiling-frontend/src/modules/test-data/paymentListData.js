export const payments = [
  {
    feeExempt: false,
    paymentCategory: 1,
    processedAmount: 7,
    submittedAmount: 7,
    serviceIdentifier: 0,
    transactionDate: null,
  },
  {
    feeExempt: true,
    paymentCategory: 3,
    processedAmount: 7,
    submittedAmount: null,
    serviceIdentifier: 0,
    transactionDate: null,
  },
  {
    feeExempt: false,
    paymentCategory: 3,
    processedAmount: 7,
    submittedAmount: 7,
    serviceIdentifier: 0,
    transactionDate: null,
  },
  {
    feeExempt: false,
    paymentCategory: 3,
    processedAmount: null,
    submittedAmount: 7,
    serviceIdentifier: 0,
    transactionDate: null,
  },
];
