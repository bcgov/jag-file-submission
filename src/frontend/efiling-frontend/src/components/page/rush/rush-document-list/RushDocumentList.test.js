import React from "react";
import { fireEvent, render } from "@testing-library/react";

import RushDocumentList from "./RushDocumentList";
import { getDocumentsData } from "../../../../modules/test-data/documentTestData";

describe("RushConfirmation Component", () => {
  const documents = getDocumentsData();

  test("Matches the snapshot", () => {
    const onDeleteFile = jest.fn();

    const { asFragment } = render(
      <RushDocumentList files={documents} onDeleteFile={onDeleteFile} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test("Download button is clickable", () => {
    // TODO: test real functionality when implemented
    // eslint-disable-next-line no-console
    console.log = jest.fn();
    const onDeleteFile = jest.fn();
    const { getByText } = render(
      <RushDocumentList files={documents} onDeleteFile={onDeleteFile} />
    );

    const fileDownloadButton = getByText(documents[0].name);
    fireEvent.click(fileDownloadButton);

    // eslint-disable-next-line no-console
    expect(console.log.mock.calls[0][0]).toBe("download");
  });

  test("Remove link onClick", () => {
    const onDeleteFile = jest.fn();
    const { getAllByText } = render(
      <RushDocumentList files={documents} onDeleteFile={onDeleteFile} />
    );

    const links = getAllByText("Remove");
    expect(links.length).toBeGreaterThan(0);

    fireEvent.click(links[0]);
    expect(onDeleteFile).toBeCalled();
  });

  test("Remove link onKeyDown Enter", () => {
    const onDeleteFile = jest.fn();
    const { getAllByText } = render(
      <RushDocumentList files={documents} onDeleteFile={onDeleteFile} />
    );

    const links = getAllByText("Remove");
    expect(links.length).toBeGreaterThan(0);

    fireEvent.keyDown(links[0], {
      key: "Enter",
      keyCode: "13",
    });
    expect(onDeleteFile).toBeCalled();
  });

  test("Remove link onKeyDown Tab", () => {
    const onDeleteFile = jest.fn();
    const { getAllByText } = render(
      <RushDocumentList files={documents} onDeleteFile={onDeleteFile} />
    );

    const links = getAllByText("Remove");
    expect(links.length).toBeGreaterThan(0);

    fireEvent.keyDown(links[0], {
      key: "Tab",
      keyCode: "9",
    });
    expect(onDeleteFile).not.toBeCalled();
  });
});
