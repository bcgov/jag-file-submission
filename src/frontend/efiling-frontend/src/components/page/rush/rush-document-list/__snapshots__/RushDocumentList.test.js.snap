// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RushConfirmation Component Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="ct-file-list "
  >
    <div
      class="header"
    >
      <span
        class="d-none d-lg-inline col-lg-6"
      >
        Document Name
      </span>
      <span
        class="d-none d-lg-inline col-lg-3"
      >
        Action
      </span>
    </div>
    <ul>
      <li>
        <span
          class="file-href col-md-12 col-lg-6"
          data-test-id="uploaded-file"
          role="button"
          tabindex="0"
        >
          <svg
            color="#FCBA19"
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(252, 186, 25);"
            viewBox="0 0 24 24"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm2 16H8v-2h8v2zm0-4H8v-2h8v2zm-3-5V3.5L18.5 9H13z"
            />
          </svg>
          file name 1
        </span>
        <span
          class="col-md-5 d-lg-none"
        >
          Statutory Fee:
        </span>
      </li>
      <li>
        <span
          class="file-href col-md-12 col-lg-6"
          data-test-id="uploaded-file"
          role="button"
          tabindex="0"
        >
          <svg
            color="#FCBA19"
            fill="currentColor"
            height="30"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(252, 186, 25);"
            viewBox="0 0 24 24"
            width="30"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 2H6c-1.1 0-1.99.9-1.99 2L4 20c0 1.1.89 2 1.99 2H18c1.1 0 2-.9 2-2V8l-6-6zm2 16H8v-2h8v2zm0-4H8v-2h8v2zm-3-5V3.5L18.5 9H13z"
            />
          </svg>
          file name 2
        </span>
        <span
          class="col-md-5 d-lg-none"
        >
          Statutory Fee:
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
